src/integration/kotlin/uk/gov/dwp/dataworks/egress/IntegrationTests.kt

```diff
@@ -177,10 +184,26 @@ class IntegrationTests: StringSpec() {
                     logger.info("$file doesn't exist")
                     delay(2000)
                 }
-                
+
                 file.readText() shouldBe sourceContents
             }
         }
+
+        "It should have pushed metrics" {
+            delay(50000)
```

Comment:
I would just poll repeatedly until the metrics appear (with a 50 second timeout) rather than having a fixed 50s delay. The `verifyEgress` method in this class does something similar for reference.